pkg/util/objectpool/objectpool.go

```diff
@@ -158,6 +163,41 @@ func (p *Pool) slowGet(ctx context.Context) (PoolObject, error) {
 	}
 }
 
+type (
+	separatedKey struct{}
+)
+
+func SetSeparatedKey(ctx context.Context, value string) context.Context {
+	return context.WithValue(ctx, separatedKey{}, value)
+}
+
+func GetSeparatedKey(ctx context.Context) string {
+	if value, ok := ctx.Value(separatedKey{}).(string); ok {
+		return value
+	} else {
+		panic("it must specify the separate key by call func `SetSeparatedKey`")
+	}
+}
+
+// Get returns an object from the pool,
+//
+// if there's an exists single, it will try to get an available object;
+// if there's no exists single, it will create a one and try to get an available object;
+func (m *MultiPool) Get(ctx context.Context) (PoolObject, error) {
+	var value interface{}
+	var ok bool
+	key := GetSeparatedKey(ctx)
+	if value, ok = m.pools.Load(key); !ok {
+		m.lock.Lock()
+		defer m.lock.Unlock()
+		if value, _ = m.pools.Load(key); !ok {
+			value = NewWithSpec(m.spec, ctx)
+			defer m.pools.Store(key, value)
+		}
+	}
```

Comment:
please use `sync.Map.LoadOrStore` to refactor.

Reply:
I tried, but func `LoadOrStore` means that a pool will be created every time, and when the `initNum` of the pool's spec is non-zero, it will create some unused connections at the same time, then these connections need to be closed if  it is load instead of store. 
'LoadOrStore' cannot achieve an effect like this
func LoadOrStore(key interface{}, value func () interface{}) {
 if key exists  {
   return old;
 } else {
  newValue = value()
  store newValue
  .....
}
}

Reply:
We can do it like this, though there's still possibility that an unnecessary object being created, but it should be fine now.

```suggestion
	if value, ok = m.pools.Load(key); !ok {
	        value = NewWithSpec(m.spec, ctx)
	        value, _ = m.pools.LoadOrStore(key, value)
	}
```

Reply:
In the case of concurrency, more than one `Pool`  woule be created by L191, and the `initnum` of these pools may be 100、200, etc., and so many `PoolObject`  instances that will not be used will be created. 
L192, the variable `value` is reassigned, It is not good for the `PoolObjects` that need to explicitly call `destroy` or `pool.Close`, such as the underlying connection, variables will be GC, but the connection still exists.
Based on the above reasons, I have a tried like the follow before deciding to use `sync.Mutex`, but it have a problem：`newValue.Close` will always be called.
```
if value, ok = m.pools.Load(key); !ok {
	       newValue = NewWithSpec(m.spec, ctx)
	       if value, ok = m.pools.LoadOrStore(key, newValue) ; ok{
                  newValue.Close()
               }
}
```

Reply:
I checked the code, this has not been updated.

Reply:
As I said, using `sync.Map` seems inappropriate in this case.
On the one hand it creates some unused object and needs to call `Destroy`, on the other hand, as shown in the code example, `LoadOrStore(key, value any) (actual any, loaded bool)`, **`loaded` always returns true, and `newValue.Close` always be called**. Of course,  we could solve this bug like the following code (use object equals instead of loaded):
```
if value, ok = m.pools.Load(key); !ok {
	       newValue = NewWithSpec(m.spec, ctx)
	       if value, _= m.pools.LoadOrStore(key, newValue) ; value != newValue  {
                  newValue.Close()
               }
}
```
For the reasons, I think it would be better to use `sync.Mutex` instead of `LoadOrStore`.